go/protocol/chat1/extras.go

```diff
@@ -1019,7 +1019,7 @@ func (p *Pagination) String() string {
 
 // FirstPage returns true if the pagination object is not pointing in any direction
 func (p *Pagination) FirstPage() bool {
-	return p == nil || (len(p.Next) == 0 && len(p.Previous) == 0)
+	return p == nil || (len(p.Next) == 0 && len(p.Previous) == 0 && !p.Last)
```

Comment:
I'm curious why this is there, seems wrong.

Reply:
might not be necessary but it seemed wrong to say it was the first page if that was set, if we respond to a query with no results `Next` and `Previous` are not set but `Last` is, so i don't think `FirstPage` should return `true` in that case

Reply:
This is only really used when determining if the `Pagination` is trying to get the first page. It is probably irrelevant what we do with `Last` here, but in the interest of just making the minimal change we could just leave it alone.